### PR TITLE
Added Dockerfile and entrypoint script

### PR DESCRIPTION
**Pull request by scanon**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+FROM ubuntu:14.04
+MAINTAINER Shane Canon <scanon@lbl.gov>
+
+#
+# Install build dependencies and openjdk-8
+#
+RUN apt-get update && apt-get -y install git ant make gcc wget software-properties-common unzip && \
+   add-apt-repository -y ppa:openjdk-r/ppa && \
```
Comment (MrCreosote) [inline Dockerfile:8]: why gcc, make, unzip? They should be installed automatically, right?
Actually, why gcc at all?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   apt-get update && \
+   apt-get -y install openjdk-8-jdk
+
+# Install jars and Jetty
+RUN \
+   mkdir /src && cd /src && git clone https://github.com/kbase/jars  && \
+   apt-get -y install unzip  && \
```
Comment (MrCreosote) [inline Dockerfile]: Already installed unzip

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ADD . /src/auth2proto
+
+RUN \
+    export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/ && \
+    cd /src/auth2proto && \
+    ant compile && \
+    ant buildwar
+
+#   24  /usr/lib/jvm/java-8-openjdk-amd64/bin/java -DSTOP.PORT=8079 -DSTOP.KEY=foo -jar /src/jetty-distribution-9.3.11.v20160721/start.jar  
+ENV KB_DEPLOYMENT_CONFIG /config/deployment.cfg
```
Comment (MrCreosote) [inline Dockerfile:28]: Where is deployment config coming from?

**Comment by MrCreosote** [comment]
Oh, also, can you add the commands to build and run an image into the readme?

File: Dockerfile (hunk 2 above)
Comment (scanon) [reply Dockerfile]: Opps meant to remove that.  I moved it up.

File: Dockerfile (hunk 1 above)
Comment (scanon) [reply Dockerfile:8]: That could probably be omitted.

File: Dockerfile (hunk 3 above)
Comment (scanon) [reply Dockerfile:28]: It gets volumed mapped in

**Comment by MrCreosote** [comment]
One other ? - this always runs master. I assume we want to run different branches or tags depending on the deploy env - how does that work?

**Comment by scanon** [comment]
You would just checkout that branch before doing the docker build.

**Comment by scanon** [comment]
I think I addressed the feedback.  Take a look.

**Comment by MrCreosote** [comment]
It still seems link wget, unzip, and software-props aren't needed, they come with 14.04

>  (this assumes mongo and nginx are already configured)

Doesn't mongo need to be running inside the docker container? Nginx can be on the host IIUC.

**Comment by kkellerlbl** [comment]
mongo can probably be running anywhere, as long as the auth2 service container can reach that host.  If mongo is in the auth2 container then configure it to talk to localhost; if it's running in a different container on the same Docker host, make sure the links between containers are configured (Shane has probably done this in docker-compose for next); if it's a host somewhere else, make sure the auth2 container can resolve DNS for that name and talk to the mongo port on that host.

nginx can also run in a different container; that's likely how next is configured.

**Comment by MrCreosote** [comment]
This repo is obsolete, please move the PR to https://github.com/kbase/auth2

**Pull request closed without merging**